### ISSUE TITLE
Dispatch returns undefined in certain cases.

### ISSUE THREAD

**Issue by tappleby** [acdlite/redux-batched-updates#1]
I brought up this issue in rackt/redux#481. 

Im not sure if this can be solved from a code point of view but it might be worth documenting the issue with dispatch result being undefined in certain cases.
- middleware needs to come after thunk, promise or any middleware where you expect a value to be returned.
- with non-thunk actions you will not be able to access the return value.

Perhaps worth listing [redux-batched-subscribe](https://github.com/tappleby/redux-batched-subscribe) as an alternative.

**Comment by peteruithoven** [comment]
I agree. Especially since @tappleby removed redux-batched-updates from the Redux ecosystem docs: https://github.com/rackt/redux/commit/dee3f07f1e81ed2a49dc8a599f1ed8952c9dc8c2
Which is confusing because most of the issues and talk mentions redux-batched-updates.
